smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/AbstractMediator.java

```diff
@@ -65,13 +117,48 @@ public void initialize(Object bean) {
                     try {
                         return this.configuration.getMethod().invoke(bean, args);
                     } catch (Exception e) {
+                        e.printStackTrace();
                         throw ex.processingException(configuration.methodAsString(), e);
                     }
                 };
             }
         }
     }
 
+    @SuppressWarnings("unchecked")
+    protected <T> T invoke(Message<?> message) {
+        try {
+            Objects.requireNonNull(this.invoker, msg.invokerNotInitialized());
+            Object[] objects = invocationParameters.apply(message);
+            return (T) this.invoker.invoke(objects);
+        } catch (RuntimeException e) { // NOSONAR
+            log.methodException(configuration().methodAsString(), e);
```

Comment:
Should this change to the exception type that logs a message and rethrows?

Reply:
Do you mean having a specific exception class that logs on creation? 

Here we log and rethrow to have a trace of the exception if the system recovers from the failure.